ionoscloud/resource_server.go

```diff
@@ -917,60 +930,59 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	// Volume stuff
 	if d.HasChange("volume") {
-		bootVolume := d.Get("boot_volume").(string)
-		_, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, dcId, d.Id(), bootVolume).Execute()
-		logApiRequestTime(apiResponse)
-
-		if err != nil {
-			volume := ionoscloud.Volume{
-				Id: &bootVolume,
-			}
-			_, apiResponse, err := client.ServersApi.DatacentersServersVolumesPost(ctx, dcId, d.Id()).Volume(volume).Execute()
-			logApiRequestTime(apiResponse)
-			if err != nil {
-				diags := diag.FromErr(fmt.Errorf("an error occured while attaching a volume dcId: %s server_id: %s ID: %s Response: %s", dcId, d.Id(), bootVolume, err))
-				return diags
-			}
-
-			// Wait, catching any errors
-			_, errState = cloudapi.GetStateChangeConf(meta, d, apiResponse.Header.Get("Location"), schema.TimeoutCreate).WaitForStateContext(ctx)
-			if errState != nil {
-				diags := diag.FromErr(fmt.Errorf("an error occured while waiting for a state change for dcId: %s server_id: %s ID: %s %w", dcId, d.Id(), bootVolume, err))
-				return diags
-			}
-		}
 
 		properties := ionoscloud.VolumeProperties{}
+		inlineVolumeIds := d.Get("inline_volume_ids")
+
+		if inlineVolumeIds != nil {
+			inlineVolumeIds := inlineVolumeIds.([]interface{})
+			for i, volumeId := range inlineVolumeIds {
+				volumeIdStr := volumeId.(string)
+				volumePath := fmt.Sprintf("volume.%d.", i)
+				_, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, dcId, d.Id(), volumeIdStr).Execute()
+				logApiRequestTime(apiResponse)
+				if err != nil {
+					diags := diag.FromErr(fmt.Errorf("an error occured while getting a volume dcId: %s server_id: %s ID: %s Response: %s", dcId, d.Id(), volumeId, err))
+					return diags
```

Comment:
@cbeti-ionos not a big deal but we can `return diag.FromErr(..`

Reply:
the fewer lines of code, the better - changed!